### ISSUE TITLE
reactive.tap clarification

### ISSUE THREAD

**Issue by timothyarmes** [ReactiveKit/Bond#451]
Hi,  

In the ReadMe you state that we can do this:

````swift
button.reactive.tap
  .observe {
    print("Button tapped.")
  }  
````

That looks nice and simple, but it generates a warning since we're ignoring the returned `Disposable`. I find that I need to do this:

````swift
button.reactive.tap.observe(with: { _ in self.doSomething() }).dispose(in: bag)
````

Which is significantly less convenient. Am I doing something wrong?

**Comment by srdanrasic** [comment]
Hey,

There is actually a mistake it that code - it should be `observeNext`, not `observe`. I'm going to fix the readme now.

Additionally, you have a retain cycle in your code - you should probably do `weak self`.

Disposing was not needed in previous versions of Bond, but in reality you almost always want to dispose observations so the compiler will print the warning to remind you of that.

I also found such approach inconvenient so there is a better way! 😄

Instead of observing a signal, you can bind it and binding with automatically take care of the disposing and retain cycles. You can get detailed info [here](https://github.com/ReactiveKit/ReactiveKit#bindings).

In short, a preferred way to observe signals is like this:

```swift
button.reactive.tap
    .bind(to: self) { me in
        me.doSomething()
    }  
```

That code is essentially a simpler way to write this:

```swift
button.reactive.tap
    .observeNext { [weak self] _ in
        guard let me = self else { return }
        me.doSomething()
    }
    .dispose(in: bag)
```

Bond readme craves for some updates. I'll see what I can do, but going thorough ReactiveKit readme is greatly recommended as Bond is build on top of it.

Hope this helps!

**Comment by timothyarmes** [comment]
Ah yes, I should have thought of binding. Thanks.
